lib/rspec/core.rb

```diff
@@ -178,4 +177,7 @@ def self.const_missing(name)
     require MODULES_TO_AUTOLOAD.fetch(name) { return super }
     ::RSpec.const_get(name)
   end
+
+  Core::DSL.expose_globally!
+  Core::SharedExampleGroup::TopLevelDSL.expose_globally!
```

Comment:
The loading and using of configuration in this file was causing problems for me and the `config.expose_dsl_globally = true` doesn't set any state in config so this seemed like the simplest solution.  Do you have a suggested alternative that doesn't involve reverting to `config.expose_dsl_globally = true`?